src/pipe_mgr/shared/dal/dpdk/dal_init.c

```diff
@@ -126,9 +126,14 @@ int dal_enable_pipeline(bf_dev_id_t dev_id,
 			 o_filepath, so_filepath);
 		status = system(buffer);
 		if (status) {
-			LOG_ERROR("%s line:%d  Cannot generate %s file\n",
```

Comment:
Sure, will do.

Reply:
I have done more experiments, and this issue is becoming very mysterious, almost "quantom mechanical". To describe my observations, let me first introduce 2 variations of the code here:

V1:
```cpp
		status = system(buffer);
		LOG_ERROR("%s line:%d  Returned status: %d\n",
			  __func__, __LINE__, status);
		if (status) {
			LOG_ERROR("%s line:%d  Cannot generate %s file\n",
				  __func__, __LINE__, so_filepath);
			return BF_INTERNAL_ERROR;
		}
```

V2:
```cpp
		status = system(buffer);
		if (status) {
			LOG_ERROR("%s line:%d  Returned status: %d\n",
				  __func__, __LINE__, status);
			LOG_ERROR("%s line:%d  Cannot generate %s file\n",
				  __func__, __LINE__, so_filepath);
			return BF_INTERNAL_ERROR;
		}
```

- I'm using `LOG_ERROR` here instead of `LOG_TRACE`, because otherwise the log message won't show up in `p4_driver.log`. I suspect this is a bug.

Observations:
- With V1, the logged message is `Returned status: 0`, and the test will pass.
- With V2, the logged message is `Returned status: -1`, and the test will fail.

I cannot make any sense of these observations. I checked the code that `LOG_ERROR` won't modify the value of `status`. Then how could the value of `status` change depending on whether it is logged or not?

**I wonder if anyone other than me can reproduce these observations?**

If you have built the project locally, just rebuild it with V1 or V2, and run the following command:
```console
# Have to make sure these files generated previously are removed,
# or the error won't show up in V2.
rm /tmp/p4.*
python3 e2e-test/tools/test.py e2e-test/suites/pna/basicfwd
```

After it finished, check the file `e2e-test/suites/pna/basicfwd/log/bf_switchd/p4_driver.log` to see the log messages.

Reply:
is the so generated even when the return value is -1 ? The only difference i can think of in between v1 & v2 is that the 
 implicit sync operation wehenever the print executes.

Reply:
Yes, the so file is generated even when the return value is -1.

Reply:
I decided to just go with V1 for now and leave the more subtle issue of logging influencing return value for future discussion maybe in its own issue.

Reply:
I did some more experiments today, and this V1/V2 difference is not reproducible. I did observe the same behaviors in some trials, but eventually V1 also gives -1 return status.

I've adopted another workaround to solve this issue, which is to check .so file existence instead of status here. I know this is not an ideal solution, but maybe we can move on with this temporary solution and try to solve it later? What do you think?

Reply:
Some further thoughts on this. Since the .so file is later used in `rte_swx_pipeline_build_from_lib(...)`, the validity of the file is checked there.